doppelganger/doppelganger.go

```diff
@@ -309,6 +315,20 @@ func (h *handler) resetDoppelgangerStates() {
 	h.logger.Info("All Doppelganger states reset to initial detection epochs")
 }
 
+func (h *handler) recordValidatorStates(ctx context.Context) {
+	var safe, unsafe uint64
+	for _, state := range h.validatorsState {
+		if state.safe() {
+			safe++
+		} else {
+			unsafe++
+		}
+	}
```

Comment:
aren't we missing a lock here around the loop?

Reply:
good catch! I think you're right, I was sure we are locking for each slot, but it's inside the functions.

Reply:
fixed